docs/.vitepress/vitepress/components/common/vp-theme-toggler.vue

```diff
@@ -12,13 +12,60 @@ watch(
     toggleDark()
   }
 )
+
+let resolveFn: (value: boolean | PromiseLike<boolean>) => void
+const isAppearanceTransition =
+  // @ts-expect-error
+  document.startViewTransition &&
+  !window.matchMedia('(prefers-reduced-motion: reduce)').matches
```

Comment:
Using `document ` and `window` like this is not very friendly to SSR

Perhaps it can be moved into the switchTheme function?

Reply:
Done.